engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala

```diff
@@ -1381,37 +1368,57 @@ class IrToTruffle(
       application match {
         case IR.Application.Prefix(fn, Nil, true, _, _, _) =>
           run(fn)
-        case IR.Application.Prefix(fn, args, hasDefaultsSuspended, loc, _, _) =>
-          val callArgFactory = new CallArgumentProcessor(scope, scopeName)
-
-          val arguments = args
-          val callArgs  = new ArrayBuffer[CallArgument]()
+        case app @ IR.Application.Prefix(
+              fn: IR.Name.Literal,
+              args,
+              false,
+              loc,
+              _,
+              _
+            ) =>
+          val global = fn.getMetadata(GlobalNames)
+          global.map(_.target) match {
+            case Some(BindingsMap.ResolvedMethod(module, method)) =>
```

Comment:
This change is a bit controversial in a sense that by making `GlobalNames` code simple and conform to the overall approach (no self arguments) we have to compensate for that in IrToTruffle translation. It seemed like a good idea at the time but it complicated the logic slightly here.
On the upside, we don't do symbol resolution for local functions in runtime.

I can bring back the old approach by reverting the change in `GlobalNames` and removing this special handling here. It would be a tiny change and I checked that it works. So happy to get feedback on this piece!